pkg/apis/troubleshoot/v1beta2/collector_shared.go

```diff
@@ -75,6 +76,7 @@ type Run struct {
 	ImagePullPolicy    string            `json:"imagePullPolicy,omitempty" yaml:"imagePullPolicy,omitempty"`
 	ImagePullSecret    *ImagePullSecrets `json:"imagePullSecret,omitempty" yaml:"imagePullSecret,omitempty"`
 	ServiceAccountName string            `json:"serviceAccountName,omitempty" yaml:"serviceAccountName,omitempty"`
+	PodSpec            corev1.PodSpec    `json:"podSpec,omitempty" yaml:"podSpec,omitempty"`
```

Comment:
The only thing i question is whether or not we should create an entirely new collector like RunPod... Did you consider that? Do you think reusing this is better?

Reply:
I think reusing is worth it if it isn't confusing for users to discern which properties are used based on when you use `podSpec` or not. I do question that. 🤔